app/controllers/api/v1/classrooms_controller.rb

```diff
@@ -55,8 +55,39 @@ def destroy
     render json: {}
   end
 
+  def initiate_all_work_groups
+    work_groups = @classroom.work_groups.where(aasm_state: :created)
+    work_groups.update_all(aasm_state: :in_progress)
+    render json: @classroom.work_groups.where(aasm_state: :in_progress).to_json
+  end
+
+  def conclude_all_work_groups
+    work_groups = @classroom.work_groups.where(aasm_state: :in_progress)
+    # need to make sure that all the worksheets are updated before sending the emails
+    mail_worksheets_to_teacher(work_groups)
+    work_groups.each { |work_group| mail_worksheets(work_group) }
+    work_groups.update_all(aasm_state: :done)
+    render json: @classroom.work_groups.where(aasm_state: :done).to_json
+  end
+
   private
 
+  def mail_worksheets(work_group)
```

Comment:
Just wondering about this method, does it mail all the worksheets to the teacher as individual files/links? Are they all in the same email?

Reply:
this method mails to the students, not the teacher
the method that mails to the teacher is `mail_worksheets_to_teacher(work_groups)` in line 84

Reply:
My mistake, is it all sending as one email with some worksheet links?

Reply:
correct, divided by group and including the names of the students in each group